tests/test_e2e.py

```diff
@@ -506,21 +504,27 @@ def test_adfs2019_onprem_acquire_token_by_auth_code(self):
         os.getenv("LAB_OBO_CLIENT_SECRET"),
         "Need LAB_OBO_CLIENT SECRET from https://msidlabs.vault.azure.net/secrets/TodoListServiceV2-OBO/c58ba97c34ca4464886943a847d1db56")
     def test_acquire_token_obo(self):
-        config = self.get_lab_user(usertype="cloud")
-
-        config_cca = {}
-        config_cca.update(config)
-        config_cca["client_id"] = "f4aa5217-e87c-42b2-82af-5624dd14ee72"
-        config_cca["scope"] = ["https://graph.microsoft.com/.default"]
-        config_cca["client_secret"] = os.getenv("LAB_OBO_CLIENT_SECRET")
-
-        config_pca = {}
-        config_pca.update(config)
-        config_pca["client_id"] = "c0485386-1e9a-4663-bc96-7ab30656de7f"
-        config_pca["password"] = self.get_lab_user_secret(config_pca["lab_name"])
-        config_pca["scope"] = ["api://%s/read" % config_cca["client_id"]]
+        if os.getenv("LAB_OBO_CONFIDENTIAL_CLIENTID") and os.getenv("LAB_OBO_PUBLIC_CLIENTID"):
```

Comment:
* The term `client_id` defined in OAuth2 specs has long been affecting our naming convention in this code base, namely, defined as TWO words connected by underscore. Please follow such naming convention.
* This condition can go into line 504, can't it? Then the majority of this function would remain the same.

Reply:
I'll change the values of variables to have underscores in them,
Also yes you can have them in line 504, however the reason its not added there is because the source of these values are different that the source of the secret in line 504. So, if we end up adding these there, it will be difficult to understand exactly what is missing and where to go to fetch the value. Also once we have the API available, these should hopefully be gone as well.

Reply:
Glad that you take into consideration on where/how to present a helpful description. But then let's explore whether we can avoid change our code structure. Possible solutions:

1. Can two `skipUnless(...)` be chained? I haven't encountered such a need before so I don't know. Please try it out and let me know. :-)
2. Or we can always use one `skipUnless(..., """a long long multi-line description as two paragraphs""")`
3. I know at least No.2 above would work but, for the sake of completeness, even if both of them would fail, we could always choose to postpone this change until the corresponding lab api arrives. In other words, it is not worthy to alter the otherwise-optimal code structure for a not-really-helpful temporary adjustment. (In case you don't realize it, the introduction of these 2 environment variables would not give us the flexibility from calling a not-yet-available lab api, and, it would make our own testing on local machine more troublesome than before. So if you/we did not set up our local testing environment due to the troublesome setup, this change alone would make it even less likely to happen.)

Reply:
I didnt really see a simple way to just chain the `skipUnless(..)` together, we can definitely do the second option. I however think that I'll prefer keeping it this way. I just think that adding too many conditions in the skipUnless can be confusing to debug if one of these environment variables goes missing. Also about your concern, we are not really changing the existing code structure, we are just adding some things keeping the existing code structure as is.  I understand that it might be harder to setup local testing environment as there will be an additional step to set these environment variables in local testing environment but I feel its a good enough one time setup to do if doing that setup means we would not be  exposing these values in the code.

Reply:
> I didnt really **see** a simple way to just chain the skipUnless(..) together

It seems to work on my machine (TM). :-)

```python
# This is the content of foo.py
import unittest


CONDITION_A = "Y"
CONDITION_B = "Y"


class TestFoo(unittest.TestCase):

    @unittest.skipUnless(CONDITION_A == "Y", "condition A not met")
    @unittest.skipUnless(CONDITION_B == "Y", "condition B not met")
    def test_foo(self):
        raise ValueError("If you see this, this test case was run")


unittest.main()
```

And then `python foo.py -v`

Reply:
Works for me too ! Thanks for this, did not realize chaining the skipUnless in this way is possible. I have made the change, please take a look :)